docs/data/routes/docs/nextjs/data-fetching/component-level-data-fetching/en.md

```diff
@@ -3,4 +3,87 @@ name: component-level-data-fetching
 routeTemplate: ./data/component-templates/article.yml
 title: Component-level data fetching
 ---
-# Component-level data fetching
\ No newline at end of file
+# Component-level data fetching
+
+Next.js provides ability to pre-render pages at build time using [getStaticProps](https://nextjs.org/docs/basic-features/data-fetching#getstaticprops-static-generation). Expected that you will fetch all required data in `getStaticProps`.
+We provide ability to do it using [SitecorePagePropsFactory](/docs/nextjs/data-fetching/data-services).
+
+So, we can pre-render pages with data fetched on the page level. What if we want to fetch component specific data on component level? Our sample provides a simple way to do it. `SitecorePagePropsFactory` is using `ComponentPropsService`.
+
+Let's dive in details how `ComponentPropsService` is doing this magick.
+
+`ComponentPropsService` accepts next parameters:
+* componentModule - function which returns Next.js component using `componentName`. You can find it in `temp/componentFactory` (generated by `scripts/bootstrap.ts`).
+* layoutData - layout service data for your page.
+* context - SSG/SSR Next.js context.
+
+`ComponentPropsService` go through layout service data and take all renderings. In order to understand which component contains side-effects, our service takes `rendering.componentName` and uses it with `componentModule`.
+Component contains side-effect and it will be called in `ComponentPropsService` in case:
+* If we are in `SSR` mode it will look for exported `getServerSideProps` function.
+* If we are in `SSG` mode it will look for exported `getStaticProps` function.
+After finishing to execute all side-effects, it will store all the data in storage using format `{ [rendering.uid]: data }`.
+
+Let's imagine that our page contains rendering, for example it's `ContentBlock` component.
+
+We can edit `src/components/ContentBlock.tsx` and add side effects in order to fetch post which we want to render.
+`useComponentProps` helps us get access to fetched data using `rendering.uid`.
+
+```tsx
+import {
+  Text,
+  RichText,
+  Field,
+  GetServerSideComponentProps,
+  GetStaticComponentProps,
+  useComponentProps,
+} from '@sitecore-jss/sitecore-jss-nextjs';
+import { StyleguideComponentProps } from 'lib/component-props';
+
+type PostEntity = {
+  title: string;
+  body: string;
+};
+
+type ContentBlockProps = StyleguideComponentProps & {
```

Comment:
Can we show how to do this without use of `StyleguideComponentProps`, so that they don't depend on that boilerplate type?

Reply:
@nickwesselman I removed this type and added `rendering` to `ContentBlockProps`